query/src/pipelines/new/executor/executor_tasks.rs

```diff
@@ -232,10 +230,16 @@ impl ExecutorTasks {
         ExecutorTask::None
     }
 
-    pub unsafe fn push_task(&mut self, worker_id: usize, task: ExecutorTask) {
+    pub fn push_task(&mut self, worker_id: usize, task: ExecutorTask) {
         self.tasks_size += 1;
+        assert!(worker_id < self.workers_sync_tasks.len(), "out of index");
```

Comment:
maybe assert_debug is better?

Reply:
Do you think release is not needed because users use our binary directly and will not be aware of it at the code level?

Reply:
Yes.